aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    adapter_common::{PreprocessedTransaction, VMAdapter},
+    aptos_vm::AptosVM,
+    block_executor::{storage_wrapper::VersionedView, AptosTransactionOutput},
+    data_cache::{AsMoveResolver, StateViewCache, StorageAdapter},
+    logging::AdapterLogSchema,
+    move_vm_ext::MoveResolverExt,
+};
+use aptos_aggregator::{delta_change_set::DeltaChangeSet, transaction::TransactionOutputExt};
+use aptos_block_executor::{
+    executor::MVHashMapView,
+    task::{ExecutionStatus, ExecutorTask},
+};
+use aptos_logger::prelude::*;
+use aptos_state_view::StateView;
+use aptos_types::{state_store::state_key::StateKey, write_set::WriteOp};
+use move_core_types::{
+    ident_str,
+    language_storage::{ModuleId, CORE_CODE_ADDRESS},
+    vm_status::VMStatus,
+};
+use std::collections::btree_map::BTreeMap;
+
+pub(crate) struct AptosVMWrapper<'a, S> {
+    vm: AptosVM,
+    base_view: &'a S,
+}
+
+fn execute_transaction<S: MoveResolverExt + StateView>(
```

Comment:
who is calling this function is a bit difficult to follow for me at the moment.
I think that must be around the problematic part you were mentioning offline...
I am still reviewing this, just fyi, and don't make it stop you from progressing...

Reply:
It is indeed the problematic part w. the separate *_btreemap_view and *_mvhashmap_view, but this we can unify in the follow-up (e.g. as in this draft https://github.com/aptos-labs/aptos-core/commit/39f055092afc2b67afcd7990ba2bc6714b90682d#diff-c357acdc9a799c00c416262b0fd8e627e1f882f603e63d77625ae0f0fd9535fa).

However, who calls execute_transaction is indeed very convoluted. I am adding a comment to explain.